python/lsst/ip/isr/crosstalk.py

```diff
@@ -941,77 +914,117 @@ def run(self,
-                                      sourceExposure.getDetector().getName())
-                        crosstalk.subtractCrosstalk(exposure, sourceExposure=sourceExposure,
-                                                    crosstalkCoeffs=interChipCoeffs,
-                                                    minPixelToMask=self.config.minPixelToMask,
-                                                    crosstalkStr=self.config.crosstalkMaskPlane,
-                                                    isTrimmed=isTrimmed,
-                                                    backgroundMethod=self.config.crosstalkBackgroundMethod)
+
+        if crosstalk.interChip:
+            if crosstalkSources:
+                # Parse crosstalkSources: Identify which detectors we have
+                # available
+                if isinstance(crosstalkSources[0], lsst.afw.image.Exposure):
+                    # Received afwImage.Exposure
+                    sourceNames = [exp.getDetector().getName() for exp in crosstalkSources]
+                elif isinstance(crosstalkSources[0], lsst.daf.butler.DeferredDatasetHandle):
+                    # Received dafButler.DeferredDatasetHandle
+                    detectorList = [source.dataId['detector'] for source in crosstalkSources]
+                    sourceNames = [camera[detector].getName() for detector in detectorList]
                 else:
-                    self.log.warning("Crosstalk contains interChip coefficients, but no sources found!")
+                    raise RuntimeError("Unknown object passed as crosstalk sources.",
+                                       type(crosstalkSources[0]))
+
+                for detName in crosstalk.interChip:
+                    if detName not in sourceNames:
+                        self.log.warning("Crosstalk lists %s, not found in sources: %s",
+                                         detName, sourceNames)
+                        continue
+                    # Get the coefficients.
+                    interChipCoeffs = crosstalk.interChip[detName]
+
+                    sourceExposure = crosstalkSources[sourceNames.index(detName)]
+                    if isinstance(sourceExposure, lsst.daf.butler.DeferredDatasetHandle):
+                        # Dereference the dafButler.DeferredDatasetHandle.
+                        sourceExposure = sourceExposure.get()
+                    if not isinstance(sourceExposure, lsst.afw.image.Exposure):
+                        raise RuntimeError("Unknown object passed as crosstalk sources.",
+                                           type(sourceExposure))
+
+                    self.log.info("Correcting detector %s with ctSource %s",
+                                  exposure.getDetector().getName(),
+                                  sourceExposure.getDetector().getName())
+                    crosstalk.subtractCrosstalk(exposure, sourceExposure=sourceExposure,
```

Comment:
This should have a check that `exposure.getBBox()` and `sourceExposure.getBBox()` match.  I think the DECam strategy has been to pre-process the `crosstalkSource` exposures, and those may not have the same trimmed state, and if they're in a different collection, I don't think this can be made consistent in the pipeline.